packages/types/src/product/common.ts

```diff
@@ -212,29 +453,67 @@ export interface FilterableProductCategoryProps
   handle?: string | string[]
   is_active?: boolean
   is_internal?: boolean
+  /**
+   * @ignore
+   * 
+   * @privateRemark
+   * This property seems to be misplaced, as it can't filter product categories.
+   */
```

Comment:
This is similar to what we have in the core. The only argument i have is that we **filter** in categories here with the rest we **filter** out. We can think about moving this to a config, but we should do that later.